src/features/register-form/index.tsx

```diff
@@ -0,0 +1,271 @@
+    switch (intent) {
+      case 'progress':
+        dispatchToast(
+          <Toast>
+            <ToastTitle media={<Spinner size="tiny" />}>{title}</ToastTitle>
+            <ToastBody>{content}</ToastBody>
+          </Toast>,
+          { toastId: progressToastId },
+        );
+        break;
+      default:
+        dispatchToast(
+          <Toast>
+            <ToastTitle>{title}</ToastTitle>
+            <ToastBody>{content}</ToastBody>
+          </Toast>,
+          { intent, timeout },
+        );
+    }
+  };
+
+  const shippingAddress = useWatch({ control, name: 'addresses.0' });
+
+  useEffect(() => {
+    if (shippingAsBilling) {
+      setValue('addresses.1', shippingAddress, { shouldValidate: true });
+    }
+  }, [shippingAddress, setValue, shippingAsBilling]);
+
+  const onSubmit = async (data: RegisterSchema) => {
+    try {
+      setIsLoading(true);
+      notify({
+        title: 'Creating an account for you...',
+        intent: 'progress',
+        content: 'Will take a second!',
+        timeout: -1,
+      });
+
+      const response = await createCustomer(data, {
+        isDefaultShippingAddress,
+        isDefaultBillingAddress,
+        shippingAsBilling,
+      });
+
+      notify({
+        title: `Hello, ${response.body.customer.firstName}! 😄`,
+        content: 'Your account was successfully created!',
+        intent: 'success',
+        timeout: 4000,
+      });
+
+      setIsLoading(false);
+      setAuthorized(true);
+      dismissToast(progressToastId);
+
+      setTimeout(() => void navigate({ to: '/' }), 2000);
+    } catch (error) {
+      setIsLoading(false);
+      dismissToast(progressToastId);
+      if (error instanceof Error) {
+        notify({
+          title: 'Oops...',
+          content: `Something went wrong: ${error.message} Please try again. 😔`,
+          intent: 'error',
+          timeout: 4000,
+        });
+      }
+    }
+  };
+
+  return (
+    <FormProvider {...methods}>
+      <form onSubmit={(e) => void handleSubmit(onSubmit)(e)}>
+        <Label weight="semibold" size="large" className={styles.label}>
+          Personal information
+        </Label>
+
+        <InputField
+          label="Email"
+          type="text"
+          placeholder="email@example.com"
+          contentBefore={<MailRegular />}
+          name="email"
+          message={errors.email?.message}
+        />
+
+        <InputField
+          label="Password"
+          contentBefore={<KeyRegular />}
+          message={errors.password?.message}
+          contentAfter={
+            <ShowHideButton className={styles.eye} onClick={() => setShow(!show)} show={show} />
+          }
+          type={show ? 'text' : 'password'}
+          name="password"
+          placeholder="Create a strong password"
+        />
+
+        <InputField
+          label="First name"
+          placeholder="Walter"
+          name="firstName"
+          type="text"
+          contentBefore={<Person24Regular />}
```

Comment:
```suggestion
          contentBefore={<PersonRegular />}
```
prefer regular icons - see docs